.github/workflows/tag-release.yml

```diff
@@ -0,0 +1,31 @@
+name: Tag Release
+
+on:
+  pull_request:
+    types: [closed]
+
+jobs:
+  tag-release:
+    if: |
+      github.event_name == 'pull_request' &&
+      github.event.action == 'closed' &&
+      github.event.pull_request.merged == true &&
+      contains(github.event.pull_request.labels.*.name, 'release')
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get current date
+        id: date
+        run: echo "::set-output name=date::$(date --rfc-3339=date)"
+
+      - name: Checkout branch "master"
+        uses: actions/checkout@v2
+        with:
+          ref: 'master'
+
+      - name: Tag Release
+        uses: tvdias/github-tagger@v0.0.2
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          tag: server-${{ steps.date.outputs.date }}
```

Comment:
What all capabilities does this action perform? Is it just creating and pushing the tag, does it create a release from the tag (is that even something we'd want)?

Don't have any concerns with this, largely asking out of curiosity, especially thinking about what all we may want to do in part 2

Reply:
It literally just creates a tag, not releases: https://github.com/tvdias/github-tagger#commit-tagger

If we want releases, we could look at something more elaborate like https://github.com/actions/create-release
The reason I have picked this is because I'm working on the assumption that we will just create tags and maintain the changelog as a markdown file (the same workflow as we have for badge-maker).

In general, I think github releases are most useful when:
- You want to associate artefacts with a release (e.g: .exe file, .deb package, etc).
  In our case, we don't need to do this.
- You want to use the release notes as the changelog.
  In our case, the workflow we've discussed is that a bot assembles a proposed changelog and then a human reviews it (and edits it if they want). This lends itself more to maintaining the changelog in a markdown file because the changelog entry can be a PR. There isn't really a way for a bot to "PR" release notes. To use the release notes as the changelog, we'd have to either take whatever is auto-generated, or make it a manual process.

If you're worried about relying on a third-party action for this, I could look at creating the tag just using shell commands instead. Using a third-party action probably isn't really buying us very much here.

Reply:
> If you're worried about relying on a third-party action for this,

Nope, don't have any particular concerns there, especially since AIUI it's just using the virtual environment token and not one from any of our accounts.

> The reason I have picked this is because I'm working on the assumption that we will just create tags and maintain the changelog as a markdown file (the same workflow as we have for badge-maker)

Works for me :+1: